airflow/providers/mysql/operators/s3_to_mysql.py

```
@@ -0,0 +1,84 @@
+    :param mysql_extra_options: MySQL options to specify exactly how to load the data.
+    :type mysql_extra_options: Optional[str]
+    :param aws_conn_id: The S3 connection that contains the credentials to the S3 Bucket.
+    :type aws_conn_id: str
+    :param mysql_conn_id: The MySQL connection that contains the credentials to the MySQL data base.
+    :type mysql_conn_id: str
+    """
+
+    template_fields = ('s3_source_key', 'mysql_table',)
+    template_ext = ()
+    ui_color = '#f4a460'
+
+    @apply_defaults
+    def __init__(self,
+                 s3_source_key: str,
+                 mysql_table: str,
+                 *args,
+                 mysql_extra_options: Optional[str] = None,
+                 aws_conn_id: str = 'aws_default',
+                 mysql_conn_id: str = 'mysql_default',
+                 **kwargs) -> None:
+        super(S3ToMySqlTransfer, self).__init__(*args, **kwargs)
+        self.s3_source_key = s3_source_key
+        self.mysql_table = mysql_table
+        self.mysql_extra_options = mysql_extra_options or ''
+        self.aws_conn_id = aws_conn_id
+        self.mysql_conn_id = mysql_conn_id
+
+    def execute(self, context: dict) -> None:
+        """
+        Executes the transfer operation from S3 to MySQL.
+
+        :param context: The context that is being provided when executing.
+        :type context: dict
+        """
+        self.log.info('Loading %s to MySql table %s...', self.s3_source_key, self.mysql_table)
+
+        s3_hook = S3Hook(aws_conn_id=self.aws_conn_id)
+        file = s3_hook.download_file(key=self.s3_source_key)
+
+        try:
+            mysql = MySqlHook(mysql_conn_id=self.mysql_conn_id)
+            mysql.bulk_load_custom(
+                table=self.mysql_table,
+                tmp_file=file,
+                extra_options=self.mysql_extra_options
+            )
+        finally:
+            # Remove file downloaded from s3 to be idempotent.
+            os.remove(file)
```

Without any except block, it would fail silently I think

---

Yep, sure will not run,  should add `except` and logging the exception.

---

from https://docs.python.org/3/tutorial/errors.html#defining-clean-up-actions

> If an exception occurs during execution of the try clause, the exception may be handled by an except clause. If the exception is not handled by an except clause, the exception is re-raised after the finally clause has been executed.

---

That is exactly what I want. I want it to raise an exception in case of failure but the file should be deleted regardless.

---

Makes sense, got it.